service/PayTrace/PayTracePaymentServices.xml

```diff
@@ -0,0 +1,493 @@
+                city:postalAddress?.city,state:postalAddress?.stateProvinceGeoId ? postalAddress.stateProvinceGeoId.substring(postalAddress.stateProvinceGeoId.indexOf('_')+1) : null, zip:postalAddress?.postalCode]]"/>
+
+                <entity-find-one entity-name="PayTrace.PaymentGatewayPayTrace" value-field="pgpt"/> 
+                <if condition="pgpt == null"><return error="true" message="No PaymentGatewayPayTrace found with ID ${paymentGatewayConfigId}"/></if>
+                <service-call name="PayTrace.PayTracePaymentServices.payTrace#login" in-map="context" out-map="token"/>
+
+                <script><![CDATA[
+                    org.moqui.util.RestClient restClient = ec.service.rest().method(org.moqui.util.RestClient.POST)
+                        .addHeader("Authorization", "bearer ${token.token.access_token}")
+                        .addHeader("Content-Type", "application/json").jsonObject(requestMap)
+                    restClient.uri().protocol("https").host(pgpt.transactionUrl).port(443).path(pgpt.apiVersion).path("transactions").path("authorization").path("keyed").build()
+                    org.moqui.util.RestClient.RestResponse restResponse = restClient.call()
+                    
+                    if (restResponse.statusCode < 200 || restResponse.statusCode >= 500) {
+                        ec.logger.error("PayTrace API response ${restResponse.statusCode}")
+                        ec.message.addMessage("PayTrace API error")
+                        return
+                    }
+                    responseMap = restResponse.jsonObject()
+                ]]></script>
+                
+                <set field="paymentOperationEnumId" value="PgoAuthorize"/>
+
+                <service-call name="PayTrace.PayTracePaymentServices.save#Response" in-map="context" out-map="context"/>
+            </actions>
+        </service>
+        <service verb="capture" noun="Payment">
+            <implements service="mantle.account.PaymentServices.capture#Payment"/>
+            <actions>
+                <entity-find-one entity-name="mantle.account.payment.Payment" value-field="payment"/>
+                <if condition="payment == null"><return message="Payment ${paymentId} not found"/></if>
+                <set field="paymentMethod" from="payment.'mantle.account.method.PaymentMethod'"/>
+                <if condition="paymentMethod?.paymentMethodTypeEnumId != 'PmtCreditCard'">
+                    <return error="true" message="Cannot capture payment ${paymentId}, not a Credit Card payment."/>
+                </if>
+
+                <!-- if no gatewayCimId, store the PaymentMethod on CIM -->
+                <if condition="!paymentMethod.gatewayCimId">
+                    <service-call name="AuthorizeDotNet.CimServices.store#CustomerPaymentMethod"
+                        in-map="[paymentMethodId:paymentMethod.paymentMethodId, paymentId:paymentId]"/>
```

Comment:
I think you mean:
`<service-call name="PayTrace.PayTraceServices.store#CustomerPaymentMethod"
                            in-map="[paymentMethodId:paymentMethod.paymentMethodId, paymentId:paymentId, paymentGatewayConfigId:paymentGatewayConfigId]"/>`